test/e2e/prometheusagent_test.go

```diff
@@ -204,3 +209,81 @@ func testPrometheusAgentStatusScale(t *testing.T) {
 		t.Fatalf("expected scale of 2 shards, got %d", pAgent.Status.Shards)
 	}
 }
+
+func testPromAgentDaemonSetResourceUpdate(t *testing.T) {
+	//t.Parallel()
+	ctx := context.Background()
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+
+	ns := framework.CreateNamespace(ctx, t, testCtx)
+	framework.SetupPrometheusRBAC(ctx, t, testCtx, ns)
+	_, err := framework.CreateOrUpdatePrometheusOperatorWithOpts(
+		ctx, testFramework.PrometheusOperatorOpts{
+			Namespace:           ns,
+			AllowedNamespaces:   []string{ns},
+			EnabledFeatureGates: []string{"PrometheusAgentDaemonSet"},
+		},
+	)
+	require.NoError(t, err)
+
+	name := "test"
+	p := framework.MakeBasicPrometheusAgentDaemonSet(ns, name)
+
+	p.Spec.Resources = v1.ResourceRequirements{
+		Requests: v1.ResourceList{
+			v1.ResourceMemory: resource.MustParse("100Mi"),
+		},
+	}
+
+	p, err = framework.CreatePrometheusAgentAndWaitUntilReady(context.Background(), ns, p)
+	require.NoError(t, err)
+
+	pods, err := framework.KubeClient.CoreV1().Pods(ns).List(ctx, pa.ListOptions(name))
+	require.NoError(t, err)
+
+	res := pods.Items[0].Spec.Containers[0].Resources
+	require.Equal(t, res, p.Spec.Resources)
+
+	p, err = framework.PatchPrometheusAgentAndWaitUntilReady(
+		context.Background(),
+		p.Name,
+		ns,
+		monitoringv1alpha1.PrometheusAgentSpec{
+			Mode: ptr.To("DaemonSet"),
+			CommonPrometheusFields: monitoringv1.CommonPrometheusFields{
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceMemory: resource.MustParse("200Mi"),
+					},
+				},
+			},
+		},
+	)
+	require.NoError(t, err)
```

Comment:
We're adding `OnDaemonsetAdd`, which means we want to react to Daemonset changes. We want to patch daemonset, not a PrometheusAgent, right?

Reply:
Feel free to tell me if I'm missing something, but the flow we want to check is:

* PrometheusAgent is created in daemonsetMode.
* Operator creates a daemonset
* We manually change the daemonset by calling k8s API
* Operator reverts the change made by us, because it is different from the desired spec.
* We verify that the Operator reacted by our changed, by making sure things were reverted

Reply:
What @ArthurSens describes would be the correct thing to test but I think that we're not doing it anyway for the other CRDs :shrug: 
This new test is still useful but it's not directly related to this change.

Reply:
> @ArthurSens Actually this test follows the similar Prometheus test here:
> 
> https://github.com/prometheus-operator/prometheus-operator/blob/4e254f018dd0ec9624409ba429c04de192ef8db0/test/e2e/prometheus_test.go#L856
> 
> Basically I would like to test if we can update Prometheus Agent DaemonSet.

I see! Like @simonpasquier said, your change is not about updates to the Agent resource. You're adding handlers to react on daemonset changes and therefore the test will be different :)